molo/core/models.py

```diff
@@ -203,6 +345,38 @@ def get_absolute_url(self):  # pragma: no cover
     def get_parent_section(self):
         return SectionPage.objects.all().ancestor_of(self).last()
 
+    def get_effective_commenting_settings(self):
+        # return commenting settings for this section
+        if (self.commenting_state and self.commenting_state != ""):
+            return {
+                'state': self.commenting_state,
+                'open_time': self.commenting_open_time,
+                'close_time': self.commenting_close_time
+            }
+        # use the commenting settings for the parent section
+        return self.get_parent_section().get_effective_commenting_settings()
+
+    def allow_commenting(self):
+        commenting_settings = self.get_effective_commenting_settings()
+        if (commenting_settings['state'] != 'O'):  # if commenting is not open
+            now = timezone.now()
+            if (commenting_settings['state'] == 'T'):
+                # Allow commenting over the given time period
+                open_time = commenting_settings['open_time']
+                close_time = commenting_settings['close_time']
+                return open_time < now < close_time
+            if (commenting_settings['state'] == 'C' or
+                    commenting_settings['state'] == 'D'):
+                # Allow automated reopening of commenting at a specified time
+                reopen_time = commenting_settings['open_time']
+                if (reopen_time):
+                    if reopen_time < now:
+                        self.commenting_state = 'O'
+                        self.save()
+                        return True
+            return False
+        return True
```

Comment:
We may want to create a ticket to instead have this something that works asynchronously with a Celery task, I think that would potentially simplify this.